src/transformers/models/mask2former/configuration_mask2former.py

```diff
@@ -0,0 +1,334 @@
+            layers, hidden state dimensions, normalization settings, etc.
+        ignore_value (`int`, *optional*, defaults to 255):
+            Category id to be ignored during training.
+        num_queries (`int`, *optional*, defaults to 150):
+            Number of queries
+        no_object_weight (`int`, *optional*, defaults to 0.1):
+            Weight to apply to the null (no object) class.
+        class_weight (`int`, *optional*, defaults to 2.0):
+            The weight for the cross entropy loss.
+        mask_weight (`int`, *optional*, defaults to 5.0):
+            The weight for the mask loss.
+        dice_weight (`int`, *optional*, defaults to 5.0):
+            The weight for the dice loss.
+        train_num_points (`str` or `function`, *optional*, defaults to 12544):
+            Number of points used for sampling during loss calculation.
+        oversample_ratio (`float`, *optional*, defaults to 3.0):
+            Oversampling parameter used for calculating no. of sampled points
+        importance_sample_ratio (`float`, *optional*, defaults to 0.75):
+            Ratio of points that are sampled via importance sampling.
+        init_std (`float`, *optional*, defaults to 0.02):
+            The standard deviation of the truncated_normal_initializer for initializing all weight matrices.
+        init_xavier_std (`float``, *optional*, defaults to 1.0):
+            The scaling factor used for the Xavier initialization gain in the HM Attention map module.
+        layer_norm_eps (`float``, *optional*, defaults to 1e-05):
+            description
+        is_train (`boolean``, *optional*, defaults to False):
+            description
+        use_auxiliary_loss (`boolean``, *optional*, defaults to True):
+            If `True` [`Mask2FormerForUniversalSegmentationOutput`] will contain the auxiliary losses computed using
+            the logits from each decoder's stage.
+        feature_strides (`List[int]`, *optional*, defaults to [4, 8, 16, 32]):
+            Feature strides corresponding to features generated from backbone network
+        deep_supervision (`boolean``, *optional*, defaults to True):
+            description
+
+    Raises:
+        `ValueError`:
+            Raised if the backbone model type selected is not in `["swin"]`.
+
+    Examples:
+
+    ```python
+    >>> from transformers import Mask2FormerConfig, Mask2FormerModel
+
+    >>> # Initializing a Mask2Former facebook/mask2former-instance-swin-small-coco configuration
+    >>> configuration = Mask2FormerConfig()
+
+    >>> # Initializing a model (with random weights) from the facebook/mask2former-instance-swin-small-coco style configuration
+    >>> model = Mask2FormerModel(configuration)
+
+    >>> # Accessing the model configuration
+    >>> configuration = model.config
+    ```
+
+    """
+    model_type = "mask2former"
+    backbones_supported = ["swin"]
```

Comment:
I think we can remove this, since it will for instance also work for Swinv2.

People can start experimenting with different backbones